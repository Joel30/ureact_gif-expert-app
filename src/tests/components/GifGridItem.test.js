import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas del componente GifGridItem', () => {

  const title = "Un título";
  const url = "https://locahost/algo.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url}/>); 

  test('debe de mostrar el componente correctamente', () => {
    

    expect(wrapper).toMatchSnapshot();
  });

  test('debe de tener un párrafo con el title', () => {
    const p = wrapper.find('p').text().trim();

    expect(p).toBe(title);
  });

  test('debe de tener la imagen igula al url y al de los props', ()=>{
    const img = wrapper.find('img').props(); //.props().src     -or-    .prop('src')

    expect(img.src).toBe(url);
    expect(img.alt).toBe(title);
  });

  test('debe de tener animate__fadeIn', () => {
    const div = wrapper.find('div').prop('className');

    expect(div.includes('animate__fadeIn')).toBe(true);
  });
});