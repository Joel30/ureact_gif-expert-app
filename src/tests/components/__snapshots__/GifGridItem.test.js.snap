// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGridItem debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://locahost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
